### DISCUSSION TITLE
Prohibit user deletion

### DISCUSSION THREAD

**Discussion by Jonbeckas** [keycloak/keycloak#33809] [Q&A] [answered]
Hi, we would like to prohibit user deletions in some cases due to legal reasons. While our go to solution would be capturing the AdminEvent via the EventListenerProviderbut  any exceptions that are thrown there are being ignored. Is there any way to prohibit the deletion without a silent rollback?

Thanks in advance

**Comment by stianst** [comment, marked as answer]
Don't think so. Event listeners can only listen, not take action, on events. You might be able to do it with extending the store provider, but wouldn't be very elegant. Think this would really need to be added in Keycloak. Fine-grained admin permissions that we are planning on starting work on soon to make supported would be a natural place for it, problem is that it's missing a way to identify who can delete which users right now. Perhaps create an issue for that?

**Reply by Jonbeckas** [reply]
Thanks for the answer. For now I will try to extend the StorageProvider and look forward

**Reply by FlxPeters** [reply]
@stianst fine grained admin permission sound interesting. Ist there already some kind of design document? We have several permission issues in the admin console when it comes to fine grained permissions. Would be super interesting if it's worth building a sidecar app handling this permission or wait for a native feature

**Reply by ssilvert** [reply]
@FlxPeters Please let us know about your use cases.  We are preparing the design for an enhanced and supported version of Fine Grained Permissions.  Details on your use cases would be very helpful.

I suggest creating an enhancement request against the admin UI.
